t/fits.t

```diff
@@ -2,340 +2,480 @@ use strict;
+          if !$PDL::Astro_FITS_Header;
+
+        # note:
+        #   the tests do not directly test the output file,
+        #   instead they write out a file, read it back in, and
+        #   compare to the data used to create the file.
+        #   So it is more of a "self consistent" test.
+
+        subtest 'auto column names, auto column types' => sub {
+
+            my $x      = long( 1, 4, 9, 32 );
+            my $y      = double( 2.3, 4.3, -999.0, 42 );
+            my $table2 = do {
+                my $table = { COLA => $x, COLB => $y };
+                my ( $fh, $file ) = tfile;
+                wfits $table, $file;
+                rfits $file;
+            };
+
+            ok( defined $table2, "Read of table returned something" );
+            is( ref($table2),  "HASH",   "which is a hash reference" );
+            is( $$table2{tbl}, "binary", "and appears to be a binary TABLE" );
+
+            ok( exists $$table2{COLA} && exists $$table2{COLB},
+                "columns COLA and COLB exist" );
+            is( $$table2{hdr}{TTYPE1}, "COLA", "column #1 is COLA" );
+            is( $$table2{hdr}{TFORM1}, "1J",   "  stored as 1J" );
+            is( $$table2{hdr}{TTYPE2}, "COLB", "column #2 is COLB" );
+            is( $$table2{hdr}{TFORM2}, "1D",   "  stored as 1D" );
+
+            is_pdl $x, $$table2{COLA}, "COLA";
+            is_pdl $y, $$table2{COLB}, "COLB";
+
+        };
+
+        subtest 'explicit column name, auto column type' => sub {
+
+            my $x = long( 1, 4, 9, 32 );
+            my $y = double( 2.3, 4.3, -999.0, 42 );
+
+            my $table2 = do {
+                my $table = {
+                    BAR => $x,
+                    FOO => $y,
+                    hdr => { TTYPE1 => 'FOO', TTYPE2 => 'BAR' }
+                };
+                my ( $fh, $file ) = tfile;
+                wfits $table, $file;
+                rfits $file;
+            };
+
+            ok(
+                defined $table2
+                  && ref($table2) eq "HASH"
+                  && $$table2{tbl} eq "binary",
+                "Read in binary table"
+            );
+            is( $$table2{hdr}{TTYPE1}, "FOO", "column #1 is FOO" );
+            is( $$table2{hdr}{TFORM1}, "1D",  "  stored as 1D" );
+            is( $$table2{hdr}{TTYPE2}, "BAR", "column #2 is BAR" );
+            is( $$table2{hdr}{TFORM2}, "1J",  "  stored as 1J" );
+
+            is_pdl $x, $$table2{BAR}, "BAR";
+            is_pdl $y, $$table2{FOO}, "FOO";
+        };
+
+        # try out more "exotic" data types
+
+        subtest 'exotic data types' => sub {
+            my $x = byte( 12, 45, 23, 0 );
+            my $y = short( -99, 100, 0, 32767 );
+            my $c = ushort( 99, 32768, 65535, 0 );
+            my $d = [ "A string", "b", "", "The last string" ];
+            my $e = float( -999.0, 0, 0, 12.3 );
+            ##my $f = float(1,0,-1,2) + i * float( 0,1,2,-1 );
+
+            my $table2 = do {
+                my $table = {
```

Comment:
This intermediate variable that's used once is not adding value. Please just call `wfits` with that value.

Reply:
ok